src/Symfony/Component/HttpFoundation/RequestMatcher/IsJsonRequestMatcher.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpFoundation\RequestMatcher;
+
+use Symfony\Component\HttpFoundation\Exception\JsonException;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestMatcherInterface;
+
+/**
+ * Checks the Request content is valid JSON.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+class IsJsonRequestMatcher implements RequestMatcherInterface
+{
+    public function matches(Request $request): bool
+    {
+        try {
+            $request->toArray();
```

Comment:
Shouldn't this request also check the Content-Type of the request instead of just trying to decode the body ?

And a json payload is not forced to be a JSON object or array (which are the only ones supported by `toArray`), so this request matcher is actually adding more restrictions, that should be documented.

Reply:
Implementation updated.
I'm not sure about the Content-Type as you have some apps out there sending JSON without the proper content type.

Reply:
IsJsonArrayMatcher then?

Reply:
@nicolas-grekas this name is confusing. `{"foo": "bar"}` is not a JSON array. It is a JSON object. But `Request::toArray()` accepts it (as it is about turning it into a *PHP* array, and PHP arrays support both lists and maps)